src/components/modals/AddressModal.js

```diff
@@ -98,6 +126,46 @@ class AddressModalConnected extends Component<Props, State> {
     this.setState({ cryptoAddress: address })
   }
 
+  filterFioAddresses = (uri: string): void => {
+    const { useUserFioAddressesOnly, userFioAddresses, isFioOnly } = this.props
+    const { fioAddresses } = this.state
+    const fioAddressesArray = []
+
+    if (useUserFioAddressesOnly) {
+      for (const address of userFioAddresses) {
+        const addressLowerCase = address.name.toLowerCase()
+        if (uri !== '' && addressLowerCase.includes(uri.toLowerCase())) {
+          fioAddressesArray.push(address.name)
+        } else if (uri === '') {
+          fioAddressesArray.push(address.name)
+        }
+      }
+    }
+
+    if (!useUserFioAddressesOnly) {
+      for (const address in fioAddresses.addresses) {
+        if (fioAddresses.addresses[address]) {
+          const addressLowerCase = address.toLowerCase()
+          if (uri !== '' && addressLowerCase.includes(uri.toLowerCase())) {
+            if (isFioOnly && !this.checkIfDomain(address) && address.includes('@')) {
+              fioAddressesArray.push(address)
+            } else if (!isFioOnly) {
+              fioAddressesArray.push(address)
+            }
+          } else if (uri === '') {
+            if (isFioOnly && !this.checkIfDomain(address) && address.includes('@')) {
+              fioAddressesArray.push(address)
+            } else if (!isFioOnly) {
+              fioAddressesArray.push(address)
+            }
+          }
+        }
+      }
+    }
```

Comment:
if (! fioAddresses.addresses[address]) continue
if ( uri !== '' && !addressLowerCase.includes(uri.toLowerCase()) continue
if (isFioOnly && (this.checkIfDomain(address) || !address.includes('@'))) continue
fioAddressesArray.push(address)